src/main/drivers/pwm_output_dshot.c

```diff
@@ -85,8 +84,44 @@ void pwmWriteDigital(uint8_t index, uint16_t value)
     for (int i = 0; i < 16; i++) {
         motor->dmaBuffer[i] = (packet & 0x8000) ? MOTOR_BIT_1 : MOTOR_BIT_0;  // MSB first
         packet <<= 1;
+    } 
+	
+	DMA_SetCurrDataCounter(motor->timerHardware->dmaRef, MOTOR_DMA_BUFFER_SIZE);
+    DMA_Cmd(motor->timerHardware->dmaRef, ENABLE);
+}
+
+void pwmWriteProShot(uint8_t index, uint16_t value)
+{
+    if (!pwmMotorsEnabled) {
+        return;
+    }
+
+    motorDmaOutput_t * const motor = &dmaMotors[index];
+
+    if (!motor->timerHardware || !motor->timerHardware->dmaRef) {
+        return;
     }
 
+    uint16_t packet = (value << 1) | (motor->requestTelemetry ? 1 : 0);
+    motor->requestTelemetry = false;    // reset telemetry request to make sure it's triggered only once in a row
+
+    // compute checksum
+    int csum = 0;
+    int csum_data = packet;
+    for (int i = 0; i < 3; i++) {
+        csum ^=  csum_data;   // xor data by nibbles
+        csum_data >>= 4;
+    }
+    csum &= 0xf;
+    // append checksum
+    packet = (packet << 4) | csum;
+	
+    // generate pulses for Proshot
+    for (int i = 0; i < 4; i++) {
```

Comment:
Mostly only this loop is different from DShot version .. It may be better to special-case buffer filling here and keep rest of function identical ...